tasks/nc_apps.yml

```diff
@@ -2,59 +2,47 @@
 - name: "[App] - Parse the item values"
   ansible.builtin.set_fact:
     nc_app_name: "{{ item.key }}"
+    nc_app_source: "{{ item.value.source if item.value.source is defined else item.value }}"
     nc_app_cfg: "{{ item.value }}"
 
-- name: "Install and enable Nextcloud Apps"
-  when: nc_app_cfg is string
+- name: "[App] - Download, install, enable and if provided configure Nextcloud Apps"
   block:
-    - name: "[App] - Download Archive in apps folder {{ nc_app_name }}"
-      ansible.builtin.unarchive:
-        copy: false
-        src: "{{ nc_app_cfg }}"
-        dest: "{{ nextcloud_webroot }}/apps/"
-        owner: "{{ nextcloud_websrv_user }}"
-        group: "{{ nextcloud_websrv_group }}"
-        creates: "{{ nextcloud_webroot }}/apps/{{ nc_app_name }}"
-      when: nc_app_cfg is not none
+    - name: "[App] - Verify the yaml declaration for the app \"{{ nc_app_name }}\""
+      ansible.builtin.assert:
+        that:
+          - (nc_app_cfg.source is defined) and (nc_app_cfg.source is string)
+        msg: "{{ nc_app_name }} is not well declared."
+      when: nc_app_cfg is not string
 
-    - name: "[App] - Enable the application {{ nc_app_name }}"
+    - name: "[App] - Install from the official app-store app: \"{{ nc_app_name }}\""
       become_user: "{{ nextcloud_websrv_user }}"
       become_flags: "{{ ansible_become_flags | default(omit) }}"
       become: true
-      ansible.builtin.command: php occ app:enable "{{ nc_app_name }}"
+      ansible.builtin.command: php occ app:install "{{ nc_app_name }}"
       args:
         chdir: "{{ nextcloud_webroot }}"
-      when: nc_app_cfg is not none
-
-- name: "[App] - Install Apps"
-  when: nc_app_cfg is mapping
-  block:
-    - name: "[App] - Verify the app's yaml declaration"
-      ansible.builtin.assert:
-        that:
-          - (nc_app_cfg.source is defined) and (nc_app_cfg.source is string)
-        msg: "{{ nc_app_name }} is not well declared."
+      when: (nc_app_source is string) and (nc_app_source | length == 0)
```

Comment:
If nc:app:source is empty, this means there is now download-link provided and app is tried to be downloaded and installed form official app-store.

Reply:
So yes, your understanding is correct. The distinction of whether or not to download is whether or not a download-link is provided. Otherwise use the official store. This way both functionalities are possible without the need to configure by the user a further setting that just distinguishes between the two cases. In my test-setup I do not even use "source", its not there and app's are fetched from the official-store.

Reply:
Ok, that's a good use case. 👍 
I think it could summarize  the condition here by something like
`nc_app_cfg.source is not defined`
Hence, the user doesn't have to specify the `source` key at all, and the variable `nc_app_source` doesn't have to be created.

Reply:
Your suggestion does not work, I was fiddling around with this until the desired result came out. This one works _without_ having the user to specify "source". AND it works if the user specifies "source". Both variants are captured and I have both in the README. I tested it once again yesterday.

Reply:
Ho I think I understand  now as I have reviewed the readme yesterday for another PR.
I saw that initial use case uses empty strings to state to enable apps or a source strings to Install for it.
So as you have iterated over this, you are now changing the behaviour when the user specify an empty string to Install instead  of enable and kept the source key for deploying from an archive.
Am I correct this time ? 😄